p2p/host/routed/routed.go

```diff
@@ -67,6 +68,43 @@ func (rh *RoutedHost) Connect(ctx context.Context, pi pstore.PeerInfo) error {
 		}
 	}
 
+	// Issue 448: if our address set includes routed specific relay addrs,
+	// we need to make sure the relay's addr itself is in the peerstore or else
+	// we wont be able to dial it.
+	for _, addr := range addrs {
+		_, err := addr.ValueForProtocol(circuit.P_CIRCUIT)
+		if err != nil {
+			// not a relay address
+			continue
+		}
+
+		if addr.Protocols()[0].Code != ma.P_P2P {
+			// not a routed relay specific address
+			continue
+		}
+
+		relay, _ := addr.ValueForProtocol(ma.P_P2P)
+
+		relayID, err := peer.IDFromString(relay)
+		if err != nil {
+			log.Debugf("failed to parse relay ID in address %s: %s", relay, err)
+			continue
+		}
+
+		if len(rh.Peerstore().Addrs(relayID)) > 0 {
+			// we already have addrs for this relay
+			continue
+		}
+
+		relayAddrs, err := rh.findPeerAddrs(ctx, relayID)
+		if err != nil {
+			log.Debugf("failed to find relay %s: %s", relay, err)
+			continue
+		}
+
+		rh.Peerstore().AddAddrs(relayID, relayAddrs, pstore.AddressTTL)
```

Comment:
This should use the temp TTL.

Reply:
The rationale wass to keep it around long enough to do all the resolutions in the address set, which may take more than 10s because of potentially multiple `findPeers`.
Also, it would be nice to avoid having to re-lookup for multiple dials to the same relay.

Reply:
will make it use the temp ttl for now, and see if this causes problems.